src/ui/base-stats-overlay.ts

```diff
@@ -0,0 +1,124 @@
+import type { InfoToggle } from "../battle-scene";
+import { TextStyle, addTextObject } from "./text";
+import { addWindow } from "./ui-theme";
+import * as Utils from "../utils";
+import i18next from "i18next";
+import { globalScene } from "#app/global-scene";
+
+export interface BaseStatsOverlaySettings {
+    scale?:number; // scale the box? A scale of 0.5 is recommended
+    x?: number;
+    y?: number;
+    /** Default is always half the screen, regardless of scale */
+    width?: number;
+}
+
+const HEIGHT = 120;
+const BORDER = 8;
+const GLOBAL_SCALE = 6;
+const shortStats = [ "HP", "ATK", "DEF", "SPATK", "SPDEF", "SPD" ];
+
+export default class BaseStatsOverlay extends Phaser.GameObjects.Container implements InfoToggle {
```

Comment:
```suggestion
export class BaseStatsOverlay extends Phaser.GameObjects.Container implements InfoToggle {
```
`default` shouldn't be used for exports (I'm aware there are other files that do this too but they shouldn't either).